app/middleware/resolve_service.js

```diff
@@ -1,17 +1,46 @@
+'use strict'
+
+// NPM modules
 const _ = require('lodash')
-const {renderErrorView} = require('../utils/response.js')
-const {isADirectDebitAccount} = require('../services/clients/direct_debit_connector_client')
 
-function notAuthorised (req, res) {
+// Local modules
+const { renderErrorView } = require('../utils/response.js')
+const { isADirectDebitAccount } = require('../services/clients/direct_debit_connector_client')
+
+const notAuthorised = (req, res) => {
   return renderErrorView(req, res, 'You do not have the rights to access this service.', 403)
 }
 
-/**
- * This middleware resolves the current service in context
- *
- */
-module.exports = function (req, res, next) {
-  const externalServiceId = req.params.externalServiceId
+const getServiceFromUserByExternalId = (req, externalServiceId) => {
+  return _.get(req.user.serviceRoles.find(serviceRole => serviceRole.service.externalId === externalServiceId), 'service')
+}
+
+const gatewayAccountType = req => {
+  let hasDDAccount = false
+  let hasCardAccount = false
+
+  if (req.service.gatewayAccountIds) {
+    req.service.gatewayAccountIds.forEach(account => {
+      if (isADirectDebitAccount(account)) {
+        hasDDAccount = true
+      } else {
+        hasCardAccount = true
+      }
+    })
+  }
```

Comment:
How about 
```
hasDDAccount = req.service.gatewayAccountIds.find(gatewayAccountId => isADirectDebitAccount(gatewayAccountId)) !== undefined
hasCardAccount = req.service.gatewayAccountIds.find(gatewayAccountId => !isADirectDebitAccount(gatewayAccountId)) !== undefined

Reply:
Did you see a refactored this in the third commit — c082763?

Reply:
oh sorry, yeah. 

`DDorNot, here I come, you can't hide`

Reply:
I still think that could be improved later, `hasDD` is not actually right, because it’s more like `isOnlyDD`. I did start refactoring that but it would have meant changes to loads of controllers, views and tests and I thought better off doing that seperately.